torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -301,7 +301,7 @@ void runTestKernel(
   // allocated here from the subgraph could be, and very likely are, different
   // from I/O expected by the generated CUDA kernel.
   for (auto& input : inputs) {
-    kernel_args.push(input, outputs[0].sizes());
+    kernel_args.push(input, outputs[0].sizes()); // why is outputs[0] special?
```

Comment:
I believe this is simply a limitation as right now we only support fusions where output sizes match. @jjsjann123